src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -354,7 +354,9 @@ class EditorObjectSelectionWindow final : public Window
         switch (widgetIndex)
         {
             case WIDX_CLOSE:
-                WindowClose(*this);
+                if (!(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER) && !CheckObjectSelection())
```

Comment:
This bit concerns me. There are four(!) situations where this code is run:
1. In the Scenario Editor
2. In the Track Designer
3. In the Track Designs Manager
4. In a regular game

What I’m afraid of is that this breaks some of those. The original issue also mentions closing via hotkey, which would also need to be tested.

Alternatively, the following code might have fewer side effects (needs testing though):
```

```
```suggestion
                if ((gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER) && !CheckObjectSelection())
                {
                    return;
                }
                WindowClose(*this);
```

Reply:
When trying this code, the game crashes when you close the track design menu with an object selected.

Reply:
Should I revert the changes?

Reply:
> here are four(!) situations where this code is run: 
>     1. In the Scenario Editor
>     2. In the Track Designer
>     3. In the Track Designs Manager
>     4. In a regular game
> What I’m afraid of is that this breaks some of those.

For the scenario editor, it should also be run.  
(Otherwise it lets you for example deselect all park entrances and then allow you to create a park without a working entrance.)